server/conf/docker/docker-compose-dev.all.debug.yml

```diff
@@ -4,9 +4,11 @@
 version: "3.8"
 services:
   cms:
-    image: taccwma/core-cms:ff737a3
+    image: taccwma/core-cms:latest
     volumes:
       - ../cms/secrets.py:/code/taccsite_cms/secrets.py
+      - ../cms/settings_local.py:/code/taccsite_cms/settings_local.py
+      - ../cms/settings_custom.py:/code/taccsite_cms/settings_custom.py
```

Comment:
In Production the :ro ensures that when a new portal VM is provisioned and configured if any of the placed files are forgotten (secrets, settings_local) the portal's docker engine will not generate an empty directory and try to mount said empty dir, docker will instead complain about the missing files - which is what we want. You may still want the mount to fail in local dev if the files are not present (ie - were not curled or copied as needed)? Its a quick way of knowing you missed something. Thoughts?

Reply:
Sounds fine, but adding `:ro` does not have that effect for me locally. I added them, ran `make stop`, deleted my existing settings & secrets files, ran `make start`, then saw empty directories added. Is my test flawed?

| b11800ab |
| - |
| <img width="960" alt="b11800ab" src="https://github.com/user-attachments/assets/619a8cb2-62d9-48ed-9bbb-e3f5d163047d"> |

Reply:
**B.** I tried using verbose syntax, so I could change the mount type to `bind`. This change, with `read_only: true` worked (caused error if file is missing).

| 59a72f8e |
| - |
| <img width="960" alt="59a72f8e" src="https://github.com/user-attachments/assets/1fb64fa7-afe7-41dd-87df-c4ad7c669d0e"> |

Reply:
**C.** I investigated again the other method I'd read, a named volume with `external: true`. It requires manually adding a volume **first** and an equally verbose docker-compose file change. And I don't think it will work (because how would container tell CMS where file is).

<details><summary>Instructions (Untested)</summary>

> 1. First, create a named volume manually (if it doesn't already exist):
>    ```sh
>    docker volume create my_volume
>    ```
> 2. Then, in your Docker Compose file, you can use this volume and mark it as external:
>     ```yml
>     services:
>       your_service:
>         image: your_image
>         volumes:
>           - my_volume:/path/in/container/file.txt
>     
>     volumes:
>       my_volume:
>         external: true
>     ```

</details>

Reply:
**D.** Move these settings and secrets files into a `server/conf/cms/settings` — one directory volume instead of many file volumes. [Internal discussion.](https://tacc-team.slack.com/archives/G0143P1HQAK/p1733784113520199)

Reply:
@rstijerina & @taoteg, I do not want to start [Idea D](https://github.com/TACC/Core-Portal/pull/1034#discussion_r1876947385) yet.[^1]

This PR, as of latest commit 4c0fb05a, would create directories. But, anyone who would customize the CMS, is [instructed to delete them, touch the files, restart, and populate the settings as necessary](https://github.com/TACC/Core-Portal/blob/4c0fb05a/docs/customize-cms.md#basic).

Is this sufferable for now?

[^1]: Because it requires refactor in Core-CMS, clients and developers to move their settings, and Camino to update where it places CMS settings.

Reply:
How about your verbose method, that doesn't create empty dirs?

Reply:
**B.2.** Ew. Okay. **See 985e86a8.** But that causes an error if file does not exist… I thought new goal was to not create files unless necessary.

**E.** To avoid error and directory and file content, create empty files. Proposal: I add back "CMS" section to `README.md` but with new command. **See 699334fd.**

I'm fine with either, just mentioning drawbacks.

_P.S. I've been thinking… we have all these commands to run; couldn't some (like 699334fd's) be moved to a script that docker runs on `up`?_

Reply:
Also, if no rush on this, it can stay open to remind me to do [Idea D](https://github.com/TACC/Core-Portal/pull/1034#discussion_r1876947385).

Reply:
Thanks for background again. Idea D is best imo